packages/components/pie-icon-button/src/index.ts

```diff
@@ -53,17 +57,23 @@ export class PieIconButton extends LitElement implements IconButtonProps {
 
     render () {
         const {
-            disabled, size, variant, isLoading,
+            disabled, size, variant, isLoading, aria,
         } = this;
 
         // The inline SVG is temporary until we have a proper icon integration
         return html`
             <button
                 class="o-iconBtn"
+                data-test-id="pie-icon-button"
                 size="${size || 'medium'}"
                 variant="${variant || 'primary'}"
                 ?disabled="${disabled}"
-                ?isLoading="${isLoading}">
+                ?isLoading="${isLoading}"
+                aria-label="${ifDefined(aria?.label)}"
+                aria-labelledby="${ifDefined(aria?.labelledby)}"
+                aria-describedby="${ifDefined(aria?.describedby)}"
+                aria-expanded="${ifDefined(aria?.expanded)}"
+                aria-controls="${ifDefined(aria?.controls)}">
```

Comment:
I see the value/use case of adding label and expanded to the icon button, but I'm having a hard time figuring out the use cases of the labelledby, describedby, and controls and was wondering if you can share more about the reasoning ? 🥇

Reply:
Sure - this was a direct port of the aria attributes exposed to consumers in Snacks. It was a suggestion from @dandel10n 😃 

<img width="1527" alt="Screenshot 2024-10-11 at 09 11 23" src="https://github.com/user-attachments/assets/15dc8f3d-4d8f-4133-aab5-01bcc7a4294e">

Reply:
I took a look at Snacks and CW, and it seems they aren't used anywhere, so I don’t think they are useful for our use cases and my concern is that copying things over might not be the ideal solution and we should only add what’s needed until we explicitly need to do otherwise.

just for context, I'm bringing this up because I realized we have a couple of unnecessary/incorrect assignments of aria attributes in the codebase, and I will try ti refine a ticket for it next week 🤝

Reply:
It is used, in snacks iconButton in Stepper for example, the attribute is also used in custom buttons in snacks so I believe it is useful to have.

Reply:
@raoufswe is there any harm in supporting these attributes? (provided they are correct usage!)